### DISCUSSION TITLE
App Service - Java App / Wjazd na maszynke

### DISCUSSION THREAD

**Discussion by vaaalery** [AzureCommunityPL/group#154] [Pomoc]
Hej,

Hostuje na App Service (windows) aplikacje javovą, zaszła potrzeba, aby trochę zbadać co dzieje się pod spodem (zrobić heap dumpa, sprawdzić memory usage jvma itp.) i niestety przez konsole wbudowaną w azure portal (jak i w kudu) po wywołaniu komendy `tasklist` (w celu znalezienia procesu javovego) dostaje komunikat: `ERROR: The user name or password is incorrect.`  Próbuje również znaleźć id processu javovego za pomocą komendy `jps`, ale znajduje mi tylko process samego wywołania jps.

Z kolej znajduje pid javovego procesu przez kudu w zakładce process explorer i chce sprawdzić memory usage za pomocą toola o nazwie jstat jednak po wywołaniu komendy `jstat.exe -gc <pid_javovy>` dostaje informacje o tym że dany pid nie istnieje.

Myślałem o tym że wbić bezpośrednio na maszynę gdzie jest hostowana aplikacja i tam odpalić te komendy, tak więc ktoś jest mi w stanie podpowiedzieć jak mogę dobić się do tej maszyny, na którą hostuje swoją aplikacje?

Ewentualnie, prośba o zasugerowanie innej drogi rozwiązania tego problemu.

**Comment by ptrstpp950** [comment]
Używasz KUDU i tam masz PowerShell albo CMD. Adres KUDU to: twoja_nazwa.scm.azurewebsites.net

**Reply by mgrabarz** [reply]
No przecież napisał, że na Kudu ma problem :)

**Comment by kamil-mrzyglod** [comment]
Spróbuj pójść do **Diagnose and solve problems** -> **Diagnostics tools**. Tam powinieneś mieć **Collect Memory Dump** i inne narzędzia.
![image](https://user-images.githubusercontent.com/11725731/99792636-aa8cb880-2b27-11eb-857a-ab314f5ccd56.png)

**Comment by mgrabarz** [comment]
Czy to są kontenery?

Sugeruję pod jave używać linuxa lub kontenerów. Tam dostaniesz `ps` z paswordless sudo, a w kontenerze możesz testowo dodać własne ssh.

**Reply by vaaalery** [reply]
Też bym wolał linuxa, ale niestety nie jestem osobą decyzyjną w tym temacie :(

**Comment by mgrabarz** [comment]
Jeszcze jedna sugestia, gdybyś musiał zostać na windows, a problem zostanie:
- https://docs.microsoft.com/en-us/rest/api/appservice/webapps/getinstanceprocess
- https://docs.microsoft.com/en-us/rest/api/appservice/webapps/getinstanceprocessdump
- https://docs.microsoft.com/en-us/rest/api/appservice/webapps/listinstanceprocesses
